render.c

```diff
@@ -64,20 +58,52 @@ void render_frame(struct swaylock_surface *surface) {
 	if (surface->image && state->args.mode != BACKGROUND_MODE_SOLID_COLOR) {
 		cairo_set_operator(cairo, CAIRO_OPERATOR_OVER);
 		render_background_image(cairo, surface->image,
-				state->args.mode, buffer_width, buffer_height);
+			state->args.mode, buffer_width, buffer_height);
 	}
 	cairo_restore(cairo);
 	cairo_identity_matrix(cairo);
 
+	wl_surface_set_buffer_scale(surface->surface, surface->scale);
+	wl_surface_attach(surface->surface, surface->current_buffer->buffer, 0, 0);
+	wl_surface_damage(surface->surface, 0, 0, surface->width, surface->height);
+	wl_surface_commit(surface->surface);
+}
+
+void render_frame(struct swaylock_surface *surface) {
+
+	struct swaylock_state *state = surface->state;
 	int arc_radius = state->args.radius * surface->scale;
 	int arc_thickness = state->args.thickness * surface->scale;
+
+	int buffer_diameter = (arc_radius + arc_thickness) * 2;
+	int buffer_height = buffer_diameter * 2;
+
+	surface->current_buffer = get_next_buffer(state->shm,
+			surface->indicator_buffers, buffer_diameter, buffer_height);
+	if (surface->current_buffer == NULL) {
+		return;
+	}
+
+	// Move subsurface out of visible area
+	wl_subsurface_set_position(surface->subsurface, surface->width, surface->height);
```

Comment:
Hmm. The correct way to hide the subsurface is to unmap it:

```c
wl_surface_attach(surface->child, NULL);
wl_surface_commit(surface->child);
```

Reply:
D'oh. I should've known there was an easier way to do it.